### PR TITLE
FEATURE: Introduce `session:collectgarbage` command and move the garbage collection from session to sessionManager

### PR DESCRIPTION
**Pull request by mficzel**
The `session:collectgarbage` command will remove session-data and -metadata of outdated sessions identified by lastActivityTimestamp being older than inactivityTimeout. This is usually done automatically after shutdown for the percentage of requests specified in the setting `Neos.Flow.session.garbageCollection.probability`. Use this command only if you need more direct control over the cleanup intervals.

In addition the `collectGarbage` method is moved from the session to the sessionManager. As this was api before a backwards compatible implementation was left behind that can be removed with Flow 9.

**Review instructions**

The main purpose of the pr is moving the garbage collection to the session manager in preparation of further adjustments in session handling where it will likely would become problematic that the session is responsible for removing other outdated sessions.

**Checklist**

- [x] Code follows the PSR-2 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked wit `!!!` and have upgrade-instructions

### CODE REVIEW

**Comment by sorenmalling** [comment]
@mficzel will you be so awesome and add a reference to a issue?

**Comment by mficzel** [comment]
@sorenmalling this is a prerequisite of finally going trying to solve https://github.com/neos/flow-development-collection/issues/525. There is no direct issue for the pr here.

**Comment by kitsunet** [comment]
+1 by reading, need to test this still.

**Review by kitsunet — APPROVED** [review]

File: Neos.Flow/Classes/Session/Session.php [hunk 1]
```diff
@@ -621,50 +616,16 @@ public function destroy($reason = null)
...
-            }
-            if ($sessionRemovalCount >= $this->garbageCollectionMaximumPerRun) {
-                break;
-            }
-        }
-
-        $this->metaDataCache->remove('_garbage-collection-running');
-        return $sessionRemovalCount;
+        $result = $this->sessionManager->collectGarbage();
+        return is_null($result) ? 0 : $result;
```
Comment (mhsdesign) [inline Neos.Flow/Classes/Session/Session.php:628]: i have the feeling we should return `false` here @mficzel ? Otherwise the return type seems to change?

**Pull request merged**